sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -683,4 +683,14 @@ object QueryExecution {
       normalized
     }
   }
+
+  def determineShuffleCleanupMode(conf: SQLConf): ShuffleCleanupMode = {
```

Comment:
This method should just check `CLASSIC_SHUFFLE_DEPENDENCY_FILE_CLEANUP_ENABLED`. If users explicitly set it to false, we should respect it and don't clean up.